default/python/AI/empire/README.md

```diff
@@ -0,0 +1,47 @@
+# Empire info module
+
+This module contains code that responsible for holding useful information about empire objects.
+
+## Implementation details. 
+
+Files in this module wildly use encapsulation principle.
+
+Data container itself is hidden from outer scope and access to it should be provided via functions.
+This approach allows changing internal implementation and add new functionality 
+without changing code which uses this module. 
+Also, we could keуp this module as simple as required for current code.
+
+Modules usually don't have any AI dependencies, so they could be easily used from any part of AI code.
+
+Each module have fucntion to set data and functions to get data. This gives us a temporal coupling. 
+Using any getter before all setters calls will lead to unexpected behaviour. Be aware of it. 
+
+To achieve encapsulation the following is done:
+The main data structure is wrapped to the `@cache_for_current_turn`. 
+So basically you will get new value only once per turn. 
+This approach require a few lines of code it is hard to forget to make a cleanup before each turn.
```

Comment:
`require` -> `requires`

Seems like a run on sentence...  not sure what is meant. Maybe `.` after `code` ?